time_stamper_ros/src/Node.cpp

```diff
@@ -106,3 +110,16 @@ bool Node::setGpioMode() {
   }
   return true;
 }
+
+void Node::configCallback(time_stamper_ros::LedConfig &config) {
+
+  //Ignore if not initialized to avoid memory corruption
+  if (!is_initialized_) {
+    return;
+  }
+  mode_ = static_cast<LedMode>(config.mode); //Convert int to enum
+
+  pwm_subsystem_.setFrequency(config.frequency);
+  setGpioMode();
```

Comment:
hmm I guess this sets the LedMode to the GPIO driver? 
Nitpick, but for maintainability it would be better to have a comment and/or put this next to the mode_ assignment. So future Mariano or Hannes don't need to think too much about sideeffects when changing anything here.

Reply:
Yeah, this isn't perfect this way. Without looking much into the code (lack of time :( ), to me the underlying problem seems to be again, the fact that this method doesn't take the mode (`static_cast<LedMode>(config.mode)`) as an argument. Wouldn't that make more sense? Then, here you wouldn't need to use the knowledge how the mode is stored (`mode_`). Maybe you wouldn't even need the `mode_` variable if you could just directly pass it to the hardware. Generally, I'd recommend to try avoiding redundant state (and that includes hardware state in addition to memory state). Duplicating it may of course be justified by the hardware being to slow or not able to provide back the value in case it needed again somewhere later.